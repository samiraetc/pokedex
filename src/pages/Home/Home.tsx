import React, { useEffect, useState } from 'react';
import { getPokemon } from '../../api/pokeapi';
import Card from '../../components/Card/Card';
import Loading from '../../components/Loading/Loading';
import Pagination from '../../components/Pagination/Pagination';
import { Pokemon } from '../../types/typing';


const Home = () => {
    const [loading, setLoading] = useState(true);
    const [totalPages, setTotalPages] = useState(0);
    const [page, setPage] = useState(0);
    const [pokemons, setPokemons] = useState([] as any);
    const itensPerPage = 20;


    const fetchPokemons = async () => {
        try {
            const data = await getPokemon(itensPerPage, itensPerPage * page);
            setPokemons(data?.pokemons);
            setTotalPages(Math.ceil(data?.count / itensPerPage));
            setLoading(false)
        } catch (error) {
            console.log('fetchPokemons error:', error)
        }
    }

    const handleBackPage = () => {
        if (page > 0) {
            setPage(page - 1)
        }
    }

    const handleNextPage = () => {
        if (page + 1 !== totalPages) {
            setPage(page + 1)
        }
    }


    useEffect(() => {
        fetchPokemons();
    }, [page])

    return (
        <div>
            {
                !loading ? (
                    <div className="container">
                        <div className="pokemon-list-container">
                            {pokemons.map((pokemon: Pokemon) => {
                                return (
                                    <Card pokemon={pokemon} />
                                )
                            })}
                        </div>
                        <Pagination
                            page={page + 1}
                            totalPages={totalPages}
                            handleBackPage={handleBackPage}
                            handleNextPage={handleNextPage}
                        />
                    </div>
                ) : (<Loading />)
            }
        </div>
    )
}

export default Home;
