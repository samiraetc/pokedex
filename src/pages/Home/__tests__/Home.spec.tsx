import { render, configure, waitFor, getByText } from '@testing-library/react'
import { act } from 'react-dom/test-utils';
import Home, { COMPONENT_ID } from '../Home';

configure({
    testIdAttribute: 'id',
  });
  
describe('[HOME]', () => {
    test('should render Home Page', async () => {
        const {asFragment, getByTestId, getByText} =  render(<Home/>);

        waitFor(() => {
            expect(getByText('bulbasaur'));
        })
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    test('should render Home Page and click to got to next page', async () => {
        const {asFragment, getByTestId, getByText} =  render(<Home/>);

        waitFor(() => {
            expect(getByText('bulbasaur'));
            expect(getByTestId('pokedex-pagination-next')).toBeInTheDocument();
        })

        expect(asFragment()).toMatchSnapshot();
    });

});