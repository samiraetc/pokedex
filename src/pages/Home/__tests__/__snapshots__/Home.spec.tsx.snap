// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[HOME] should render Home Page 1`] = `
<DocumentFragment>
  <div
    id="pokedex-home"
  >
    <div
      id="pokedex-loading"
    >
      <img
        alt="pokebola icon"
        class="loading"
        src="default"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`[HOME] should render Home Page and click to got to next page 1`] = `
<DocumentFragment>
  <div
    id="pokedex-home"
  >
    <div
      id="pokedex-loading"
    >
      <img
        alt="pokebola icon"
        class="loading"
        src="default"
      />
    </div>
  </div>
</DocumentFragment>
`;
